### PR TITLE
drivers: adc/wifi: esp32s2/esp32c3: call to adc2_init_code_calibration

### PR DESCRIPTION
**Pull request by wmrsouza**
Add call to adc2_init_code_calibration() during adc and wifi initialization on esp32s2 and esp32c3

### CODE REVIEW

File: drivers/adc/adc_esp32.c [hunk 1]
```diff
@@ -654,6 +654,12 @@ static int adc_esp32_init(const struct device *dev)
 
 	adc_hw_calibration(conf->unit);
 
+#if CONFIG_SOC_SERIES_ESP32S2 || CONFIG_SOC_SERIES_ESP32C3
+	if (conf->unit == 1) {
```
Comment (uLipe) [inline drivers/adc/adc_esp32.c:658]: Can we put this hardcoded value behind of a define and explain its reason?

File: drivers/adc/adc_esp32.c [hunk 2]
```diff
@@ -57,6 +57,10 @@ LOG_MODULE_REGISTER(adc_esp32, CONFIG_ADC_LOG_LEVEL);
 
 #define ADC_DMA_BUFFER_SIZE	DMA_DESCRIPTOR_BUFFER_MAX_SIZE_4B_ALIGNED
 
+#if CONFIG_SOC_SERIES_ESP32S2 || CONFIG_SOC_SERIES_ESP32C3
+#define ESP32_ADC2_UNIT 1
```
Comment (sylvioalves) [inline drivers/adc/adc_esp32.c:61]: Where is it used?

Comment (wmrsouza) [reply drivers/adc/adc_esp32.c:61]: It's used on line 662 of this same file.

Comment (sylvioalves) [reply drivers/adc/adc_esp32.c:61]: Sound something isn't correct.
We do have  `ADC_UNIT_1` and `ADC_UNIT_2` available in the code. Why having `ESP32_ADC2_UNIT` at all?
The only instance where it is used is at
```c
#if CONFIG_SOC_SERIES_ESP32S2 || CONFIG_SOC_SERIES_ESP32C3
	if (conf->unit == ESP32_ADC2_UNIT) {
		adc2_init_code_calibration();
	}
#endif /* CONFIG_SOC_SERIES_ESP32S2 || CONFIG_SOC_SERIES_ESP32C3 */
```
It seems you can just trash `ESP32_ADC2_UNIT` and use `ADC_UNIT_1`?

**Review by sylvioalves — APPROVED** [review]

**Review by raffarost — APPROVED** [review]

**Pull request merged**